### DISCUSSION TITLE
OpenSSF Best Practices Badge Program

### DISCUSSION THREAD

**Discussion by marcwrobel** [keycloak/keycloak#13276] [Ideas]
The [Open Source Security Foundation (OpenSSF) Best Practices badge](https://bestpractices.coreinfrastructure.org/) is a way for Free/Libre and Open Source Software (FLOSS) projects to show that they follow security best practices. Many popular projects (such as Linux, GitLab, Prometheus, OpenSSL...) already joined this program and I think it would send a positive message to the users if Keycloak could join it too.

Is the Keycloak team aware this program exists ? Is it something you could consider ?

**Comment by abstractj** [comment]
@marcwrobel maybe some time in the future. Currently, our focus is on CNCF. See: https://github.com/keycloak/keycloak/discussions/12975

**Comment by tsaarni** [comment]
@abstractj Just clarifying a bit about the scope of this, if there is any doubt: 

It is not that Keycloak would need to apply to an organization, similar to CNCF.  It is just about entering project into the system by filling in self-evaluation form about the project. The form has questions how the project works in relation to given best practices [criteria](https://bestpractices.coreinfrastructure.org/en/criteria/0). These are commonly acknowledged best practices, so no rocket science :) The answers will be scored, which is then summarized by the "badge". Project is entered by [logging in](https://bestpractices.coreinfrastructure.org/en/login) using github account and [registering](https://bestpractices.coreinfrastructure.org/en/projects/new) a new project. Since Keycloak is in github, few of the answers are automatically filled in.

I know that some organizations evaluate and choose Open Source projects by using the best practices badge as input. I think it would be good investement of time if one of the project maintainers would take a moment to fill in the self-evaluation form. The purpose is just to communicate how Keycloak as a project thinks about these basic software development and security topics.

I don't see reason why not enter the project into the system?

**Reply by tsaarni** [reply]
Just to test this out, I just entered one of my own projects https://bestpractices.coreinfrastructure.org/en/projects/6334. It requires good knowledge of the software itself, its dependencies and about the working practices that the maintainer team is committed to.

**Reply by marcwrobel** [reply]
Here is another example for one of my project : https://bestpractices.coreinfrastructure.org/en/projects/6217.

As @tsaarni said, it requires a good knowledge of every aspects of the project. FYI it took me 3 to 6 hours to answer all the questions (it was also the opportunity to improve a lot of things in my project).

**Comment by abstractj** [comment]
@tsaarni created https://github.com/keycloak/keycloak/issues/13574 so we don't miss it.
